.github/workflows/ci.yml

```
@@ -28,9 +24,51 @@ jobs:
         run: bin/docs
       - name: Verify README
         run: bin/readme
-      - name: Run tests
-        run: bin/test
       - name: Verify gem RBIs are up-to-date
         run: bundle exec exe/tapioca gem --verify
       - name: Verify duplicates in shims
         run: bundle exec exe/tapioca check-shims
+
+  tests:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: ["2.7", "3.0", "3.1"]
+        experimental: [false]
+        gemfile:
+          - Gemfile
+          - gemfiles/Gemfile-rails-6-1
+        include:
+          - gemfile: gemfiles/Gemfile-rails-main
+            ruby: "2.7"
+            experimental: true
+          - gemfile: gemfiles/Gemfile-rails-main
+            ruby: "3.0"
+            experimental: true
+          - gemfile: gemfiles/Gemfile-rails-main
+            ruby: "3.1"
+            experimental: true
+    name: Ruby ${{ matrix.ruby }} - Gemfile ${{ matrix.gemfile }}
```

One nitpick: Do we really need to write `Gemfile` here, since the gemfile itself has that in its name
```suggestion
    name: Ruby ${{ matrix.ruby }} - ${{ matrix.gemfile }}
```